src/modules/scraper/service.ts

```diff
@@ -52,6 +58,45 @@ export class ScraperService {
     }
   }
 
+  public async publishMerkleDistributorBlocks() {
+    const chainId = this.appConfig.values.web3.merkleDistributor.chainId;
+    const configStartBlockNumber = this.appConfig.values.web3.merkleDistributor.startBlockNumber;
+    const provider = this.providers.getProvider(chainId);
+    const latestBlock = await provider.getBlock("latest");
```

Comment:
I think we can avoid fetching the latest blocks again, since they were already fetched in the `publishBlocks()` function. We can call the `getLatestBlocks()` at the beginning of the `run()` function and pass the result to `this.publishBlocks()` and `this.publishMerkleDistributorBlocks()`